src/dataExplorer/context/resultsView.tsx

```diff
@@ -109,20 +152,209 @@ export const ResultsViewProvider: FC = ({children}) => {
     persistSelectedViewOptions(DEFAULT_VIEW_OPTIONS)
   }
 
-  useEffect(() => {
-    // if parent query is re-run => decide what to reset in subquery viewOptions
+  const setViewProperties = (dataColumn: string) => {
+    // known, returned data shape. Unpivoted. Single `_value` column.
+    const smoothedData = selectedViewOptions.smoothing.applied
+    const hasSingleValueColumn = smoothedData
+
+    switch (view.properties.type) {
+      case 'xy':
+      case 'heatmap':
+      case 'scatter':
+        setView({
+          ...view,
+          properties: {
+            ...view.properties,
+            ...(smoothedData ? {yColumn: '_value'} : {yColumn: dataColumn}),
+            ...(smoothedData ? {xColumn: '_time'} : {xColumn: timeColumn}),
+          },
+        })
+        break
+      case 'histogram':
+        // These graphs only have a single xColumn to choose.
+        setView({
+          ...view,
+          properties: {
+            ...view.properties,
+            ...(hasSingleValueColumn
+              ? {xColumn: '_value'}
+              : {xColumn: dataColumn}),
+          },
+        })
+        break
+      case 'gauge':
+      case 'single-stat':
+        // These graphs are about calculating a single value.
+        // The graph unions over all `results.parsed.fluxGroupKeyUnion = []`, to try creating a single value.
+        // For the unpivoted data --> it has all field values under a single column `_value`. Then fluxGroupKeyUnion = tagKeys.
+        // For the pivoted data --> it has the data broken into different column per field. Cannot support.
+        if (!hasSingleValueColumn) {
+          dispatch(notify(trySmoothingData(view.properties.type)))
+        }
+        break
+      case 'line-plus-single-stat':
+        // Has the same issue as 'single-stat'. Read above.
+        if (!hasSingleValueColumn) {
+          dispatch(notify(trySmoothingData('single-stat')))
+        }
+        // But the line graph will still work.
+        setView({
+          ...view,
+          properties: {
+            ...view.properties,
+            ...(smoothedData ? {yColumn: '_value'} : {yColumn: dataColumn}),
+            ...(smoothedData ? {xColumn: '_time'} : {xColumn: timeColumn}),
+          },
+        })
+        break
+      case 'band':
+      case 'check':
+      case 'mosaic':
+      // TODO: Is not working at all.
+      // Unclear if there is a data query results format, in which it will work.
```

Comment:
TODO: decide what to do with the band, check, and mosaic graphs.

Reply:
Mosaic graphs were added as an intern project, and have never been highly supported and have probably been a more niche graph type. I'm in favor of removing it in this context

Reply:
Per feedback, am removing.